docker/README.md

```diff
@@ -0,0 +1,108 @@
+# Orion Docker Setup
+
+This document provides detailed instructions for building and running Orion using Docker.
+
+## Building the Image
+
+```bash
+docker build -t orion-proxy -f docker/Dockerfile .
+```
+
+## Running the Container
+
+### Basic Setup
+
+Use port mapping:
+
+```bash
+docker run -d \
+  -p 8000:8000 \
+  --name orion-proxy \
+  orion-proxy
+```
+
+For full functionality (including TCP proxy):
+
+```bash
+docker run -d \
+  -p 8000:8000 \
+  -p 8001:8001 \
+  --name orion-proxy \
+  orion-proxy
+```
+
+Alternative with host networking:
+
+```bash
+docker run -d \
+  --network host \
+  --name orion-proxy \
+  orion-proxy
+```
+
+> **Note**: Host networking may not work in all Docker environments (e.g., Docker Desktop on macOS/Windows). Use port mapping for reliable access to services.
+
+## Configuration Options
+
+### Custom Configuration File
+
+Mount your own configuration file:
+
+```bash
+docker run -d \
+  --network host \
+  -v /path/to/your/config.yaml:/etc/orion/orion-runtime.yaml \
+  --name orion-proxy \
+  orion-proxy
+```
+
+### Control Plane Setup
+
+Configure with external control plane:
+
+```bash
+docker run -d \
+  --network host \
+  -e CONTROL_PLANE_IP=<your_control_plane_ip> \
+  --name orion-proxy \
+  orion-proxy
+```
+
+### Environment Variables
+
+| Variable | Description | Default |
+|----------|-------------|---------|
+| `CONTROL_PLANE_IP` | IP address of the control plane | `127.0.0.1` |
+| `LOG_LEVEL` | Logging level (debug, info, warn, error) | `info` |
+
+## Verification
+
+### Service Health Check
+Once the container is running, verify the HTTP listener is responding.
+
+**If using port mapping (`-p 8000:8000`):**
+```bash
+curl -v http://localhost:8000/direct-response # Should return HTTP 200 with "meow! 🐱"
+```
+
+**If using host networking (`--network host`) and it's working:**
+```bash
+curl -v http://localhost:8000/direct-response # Should return HTTP 200 with "meow! 🐱"
+```
+
+### Load Balancing Test
+Test the default route that forwards to the configured backend cluster:
+
+```bash
+curl -v http://localhost:8000/ # Will attempt to proxy to backend (may timeout if backends aren't running)
+```
+
```

Comment:
It would be great if you can setup a backend cluster here. Need to change the IP and port in cluster config, ref: https://github.com/kmesh-net/orion/blob/main/orion-proxy/conf/orion-runtime.yaml#L98